compiler/test/unit/parser/linkage_location.d

```diff
@@ -212,33 +212,10 @@ extern (C++) static class LinkVisitor : SemanticTimeTransitiveVisitor
     }
 }
 
-enum Links = [
-    "C",
-    "C++",
-    "D",
-    "Objective-C",
-    "Windows",
-    "System"];
-
-static foreach (link; Links)
-{
-    @("extern("~link~")")
-    unittest
-    {
-        auto t = parseModule("test.d", "first_token extern("~link~")");
-
-        scope visitor = new LinkVisitor;
-        t.module_.accept(visitor);
-
-        assert(visitor.l.linnum == 1);
-        assert(visitor.l.charnum == 13);
-    }
-}
```

Comment:
I don't know why it's testing the location on a broken AST, the following tests look more sensible (except the 'first_token' part).

Reply:
This doesn't seem related to this PR.

Reply:
With this PR, when the parser sees a keyword used as an identifier, it skips over it (`nextToken()`) to prevent further broken error messages. This broken test relied on the fact that after encountering a declaration named `extern` with type `first_token`, it would continue with parsing the `extern` declaration anyways.